### PR TITLE
Fix #21380 - Hint the user with quoting the /R and /ad/ ##shell

### LINKED ISSUE #21380: improve /R command docs to include ""
## Description
for some regexp you need to surround them with quotations "likethis"

*Provide images, ascii-art, test files and anything that may help us understand your request*
Screen Shot 2023-02-22 at 08 19 54 AM

### PR DESCRIPTION
**Pull request by trufae**
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/core/cmd_search.c [hunk 1]
```diff
@@ -207,7 +207,7 @@ static const char *help_msg_slash_r[] = {
 };
 
 static const char *help_msg_slash_R[] = {
-	"Usage: /R", "", "search for ROP gadgets",
+	"Usage: /R", "", "search for ROP gadgets (hint: quote the command if your regex uses special chars)",
```
Comment (hamaney) [inline libr/core/cmd_search.c:210]: optional : (hint:  surround your regex with a double quotation if your regex uses special chars. eg: "regex1 | regex2" ) 

I think it is more clear.

Comment (trufae) [reply libr/core/cmd_search.c:210]: thats not correct. because its the whole command that needs to be quoted or just start the command with ""

Comment (hamaney) [reply libr/core/cmd_search.c:210]: Yes. This was before learning it yesterday. We need the word R2 command with the quote.

should render like this:
```
search for ROP gadgets (hint: quote the whole r2 command if your regex uses special chars. eg: "/R/ regx1;regx2")
```

Comment (trufae) [reply libr/core/cmd_search.c:210]: that's too large to fit in a 80x24 screen for something that affects all commands, imho is better to add a hint and put a proper explanation somewhere else in a generic way for all the commands that the user may want to use special foes

Comment (trufae) [reply libr/core/cmd_search.c:210]: i mean, all shells have special characters, and none of them reminds the user to escape the chars

Comment (AbdulAbCellera) [reply libr/core/cmd_search.c:210]: Fair enough. Would the length of you current changes allow for 2 more chars for (r2) ?

Comment (hamaney) [reply libr/core/cmd_search.c:210]: Fair enough. Would the length of you current changes allow for 2 more chars for (r2) ?

**Pull request closed without merging**